src/script/page/AppLock/AppLock.tsx

```diff
@@ -95,10 +104,17 @@ const AppLock: React.FC<AppLockProps> = ({
 
   const {current: modalObserver} = useRef(
     new MutationObserver(() => {
-      const modalInDOM = document.querySelector('[data-uie-name="applock-modal"]');
-      if (!modalInDOM) {
-        amplify.publish(WebAppEvents.LIFECYCLE.SIGN_OUT, SIGN_OUT_REASON.USER_REQUESTED);
-      }
+      let retries = 0;
+
+      const interval = setInterval(() => {
```

Comment:
Here was a race condition, where the modal entered the view after the first check of mutationObserver. This lead to the user constantly being logged out.

Reply:
can you explain a little more what is happening here? Why does the `style` changes after the modal was shown, I don't think I understand 🤔 
Why does it not happen all the time the applock is enabled?

I have the feeling there is a cleaner fix for that